### PR TITLE
Nvilla/topic/more contact types

### PR DESCRIPTION
**Pull request by NahuelVilla**
Several updates on the package.

Now we have three classes for the contacts, and I have included latex documentation.
Tests are working, though we should add some tests for the `contactPoit` class.

It is ready to be merged.

### CODE REVIEW

File: include/dynacom/contact6d.hpp [hunk 1]
```diff
@@ -88,130 +75,64 @@ class Contact6D {
...
-  const Eigen::Matrix<double, 1, 1> &uni_b() { return unilaterality_b_; }
-  const Eigen::Matrix<double, 4, 3> &fri_A() { return friction_A_; }
-  const Eigen::Matrix<double, 4, 1> &fri_b() { return friction_b_; }
-  const Eigen::Matrix<double, 3, 1> &reg_A() { return regularization_A_; }
-  const Eigen::Matrix<double, 3, 1> &reg_b() { return regularization_b_; }
-  const Eigen::Matrix<double, 6, 3> &NE_A() { return newton_euler_A_; }
-
-  const Eigen::Matrix<double, 3, 1> &appliedForce() { return contactForce_; }
-};
+// struct ContactPointSettings {
```
Comment (MaximilienNaveau) [inline include/dynacom/contact6d.hpp]: all these comments to be removed.

File: python/contact_base.cpp [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      .def("fri_b", &ContactBase::fri_b,
+           bp::return_value_policy<bp::reference_existing_object>(),
+           bp::args("self"))
+      .def("reg_A", &ContactBase::reg_A,
+           bp::return_value_policy<bp::reference_existing_object>(),
+           bp::args("self"))
+      .def("reg_b", &ContactBase::reg_b,
+           bp::return_value_policy<bp::reference_existing_object>(),
+           bp::args("self"))
+      .def("NE_A", &ContactBase::NE_A,
```
Comment (MaximilienNaveau) [inline python/contact_base.cpp]: why NE in upper case while al the rest is in lower case?

File: tests/python/test_contacts.py [hunk 3]
```diff
@@ -19,7 +19,7 @@
 
 unittest.util._MAX_LENGTH = 2000
 
-debug = False
+debug = True
```
Comment (MaximilienNaveau) [inline tests/python/test_contacts.py:22]: what does it do? is this need as default?

File: documentation/dynacom_docs.tex [hunk 4]
```diff
@@ -0,0 +1,270 @@
+%\documentclass{article}
```
Comment (MaximilienNaveau) [inline documentation/dynacom_docs.tex:1]: nice to upload the PDF, maybe a makefile would be nice, this can be another pr

**Review by MaximilienNaveau — CHANGES_REQUESTED** [review]
Great job!
Small details to work out before merging, but overall it's very nice work!

**Comment by MaximilienNaveau** [comment]
This is linked to this issue https://github.com/Gepetto/dynacom/issues/2

File: python/contact_base.cpp (hunk 2 above)
Comment (NahuelVilla) [reply python/contact_base.cpp]: Because Newton and Euler are proper names.
In the same way we write the units "meter" in lower case and "Volt" starting with a capital letter. Still, this may not apply for code. Let me know if you still think that they should be lower-case.

File: tests/python/test_contacts.py (hunk 3 above)
Comment (NahuelVilla) [reply tests/python/test_contacts.py:22]: No, the default should be False.

**Comment by NahuelVilla** [comment]
@MaximilienNaveau I guess the final commit satisfied your comments. Let me know if you agree to merge it.

**Review by MaximilienNaveau — APPROVED** [review]
Ready to be merged.
Please click on "rebase and merge"
I don't have the rights on this repos to do it.

**Comment by NahuelVilla** [comment]
It cannot be rebased due to conflicts.

**Comment by MaximilienNaveau** [comment]
ok then just merge
